tests/rptest/tests/topic_creation_test.py

```diff
@@ -185,6 +196,64 @@ def topic_is_healthy():
         swarm.stop()
         swarm.wait()
 
+    @cluster(num_nodes=6)
+    def test_cloud_topic_recreation_while_producing(self):
+        """
+        Test that we are able to recreate topic multiple times
+        """
+        self._client = DefaultClient(self.redpanda)
+
+        # scaling parameters
+        partition_count = 30
+        producer_count = 10
+
+        spec = TopicSpec(
+            partition_count=partition_count,
+            replication_factor=3,
+            cloud_topics_enabled=True,
+        )
+
+        self.client().create_topic(spec)
```

Comment:
See this comment on the same situation https://github.com/redpanda-data/redpanda/pull/28130#discussion_r2453651929